packages/sdk/src/swaps/smart-router/routes/amm-swap-route.ts

```diff
@@ -0,0 +1,208 @@
+    }
+    return swapConverter;
+  };
+
+  const getConverterContract = async (entry: string, destination: string) => {
+    if (isNativeToken(entry) || isNativeToken(destination)) {
+      if (!rbtcConverter) {
+        const chainId = await getChainId();
+        const { address, abi } = await getProtocolContract(
+          'btcWrapperProxy',
+          chainId,
+        );
+        rbtcConverter = new Contract(address, abi, provider);
+      }
+      return rbtcConverter;
+    }
+
+    return getSwapNetworkContract();
+  };
+
+  const getSwapQuoteContract = async () => {
+    if (!protocolContract) {
+      const chainId = await getChainId();
+      const { address, abi } = await getProtocolContract('protocol', chainId);
+      protocolContract = new Contract(address, abi, provider);
+    }
+    return protocolContract;
+  };
+
+  const isNativeToken = (token: string) => token === constants.AddressZero;
+
+  const validatedTokenAddress = async (token: string) => {
+    token = token.toLowerCase();
+    if (isNativeToken(token)) {
+      if (wrbtcAddress) {
+        return wrbtcAddress;
+      }
+      const chainId = await getChainId();
+      wrbtcAddress = (await getTokenContract(SupportedTokens.wrbtc, chainId))
+        .address;
+      return wrbtcAddress;
+    }
+
+    return token;
+  };
+
+  return {
+    name: 'AMM',
+    pairs: async () => {
+      if (!pairCache) {
+        const chainId = await getChainId();
+
+        const swapTokens = [
+          SupportedTokens.sov,
+          SupportedTokens.dllr,
+          SupportedTokens.mynt,
+          SupportedTokens.rbtc,
+          SupportedTokens.xusd,
+          SupportedTokens.doc,
+          SupportedTokens.fish,
+          SupportedTokens.rif,
+          SupportedTokens.bpro,
+          SupportedTokens.rusdt,
+          SupportedTokens.eths,
+          SupportedTokens.bnbs,
+          SupportedTokens.moc,
+        ];
+
+        const contracts = await Promise.all(
+          swapTokens.map(token => getTokenContract(token, chainId)),
+        );
+
+        const addresses = contracts.map(contract =>
+          contract.address.toLowerCase(),
+        );
+
+        const pairs = new Map<string, string[]>();
+
+        for (const address of addresses) {
+          const pair = addresses.filter(a => a !== address);
+          pairs.set(address, pair);
+        }
+
+        pairCache = pairs;
+      }
+
+      return pairCache;
+    },
+    quote: async (entry, destination, amount, options?, overrides?) => {
+      const baseToken = await validatedTokenAddress(entry);
+      const quoteToken = await validatedTokenAddress(destination);
+      return (await getSwapQuoteContract())
+        .getSwapExpectedReturn(baseToken, quoteToken, amount)
+        .catch(e => {
+          throw makeError(e.message, SovrynErrorCode.ETHERS_CALL_EXCEPTION);
+        });
+    },
+    approve: async (entry, destination, amount, from, overrides) => {
```

Comment:
`from` is unused

Reply:
It's part of interface signature, so can't remove them even if it's not used now. As it's approve request some of routes will require senders address to verify if amount is already approved

Reply:
Yeah, I didn't think about that.